corgi/core/models.py

```diff
@@ -839,30 +840,25 @@ def get_roots(self) -> list[ComponentNode]:
         if not self.software_build:
             return roots
         for cnode in self.cnodes.get_queryset():
-            try:
-                root = cnode.get_root()
-                if root.obj.type == Component.Type.CONTAINER_IMAGE:
-                    # TODO if we change the CONTAINER->RPM ComponentNode.type to something besides
-                    # 'PROVIDES' we would check for that type here to prevent 'hardcoding' the
-                    # container -> rpm relationship here.
-                    # RPMs are included as children of Containers as well as SRPMs
-                    # We don't want to include Containers in the RPMs roots.
-                    # Partly because RPMs in containers can have unprocesses SRPMs
-                    # And partly because we use roots to find upstream components,
-                    # and it's not true to say that rpms share upstreams with containers
-                    rpm_desendant = False
-                    for ancestor in cnode.get_ancestors(include_self=True):
-                        if ancestor.obj.type == Component.Type.RPM:
-                            rpm_desendant = True
-                            break
-                    if not rpm_desendant:
-                        roots.append(root)
-                else:
+            root = cnode.get_root()
+            if root.obj.type == Component.Type.CONTAINER_IMAGE:
+                # TODO if we change the CONTAINER->RPM ComponentNode.type to something besides
+                # 'PROVIDES' we would check for that type here to prevent 'hardcoding' the
+                # container -> rpm relationship here.
+                # RPMs are included as children of Containers as well as SRPMs
+                # We don't want to include Containers in the RPMs roots.
+                # Partly because RPMs in containers can have unprocesses SRPMs
+                # And partly because we use roots to find upstream components,
+                # and it's not true to say that rpms share upstreams with containers
+                rpm_desendant = False
```

Comment:
I know this is unrelated to your change, but could you fix this typo please? :smiley: